registry/storage/driver/gcs/gcs.go

```diff
@@ -441,35 +444,40 @@ func (w *writer) Close() error {
 
 func putContentsClose(wc *storage.Writer, contents []byte) error {
 	size := len(contents)
-	var nn int
-	var err error
-	for nn < size {
+
+	var (
+		written int
+		err     error
+	)
+
+	for written < size {
 		var n int
-		n, err = wc.Write(contents[nn:size])
-		nn += n
+		n, err = wc.Write(contents[written:size])
+		written += n
 		if err != nil {
-			break
+			return err
 		}
 	}
-	if err != nil {
-		return err
-	}
+
 	return wc.Close()
 }
 
 // Commit flushes all content written to this FileWriter and makes it
 // available for future calls to StorageDriver.GetContent and
 // StorageDriver.Reader.
 func (w *writer) Commit(ctx context.Context) error {
-	if err := w.checkClosed(); err != nil {
-		return err
+	if w.closed {
+		return fmt.Errorf("already closed")
+	} else if w.committed {
+		return fmt.Errorf("already committed")
+	} else if w.cancelled {
+		return fmt.Errorf("already cancelled")
```

Comment:
Should this be in a func called `precheck`? seems to be repeated more than couple times.

Reply:
Actually, these slightly differ from func to func (`Cancel`, `Close` and `Write` have these checks). Theyre done the same way in the S3 driver.